packages/core/core-flows/src/order/workflows/create-exchange-return-shipping-method.ts

```diff
@@ -0,0 +1,145 @@
+  function (input: {
+    exchangeId: string
+    shippingOptionId: string
+    customShippingPrice?: BigNumberInput
+  }): WorkflowData {
+    const orderExchange: OrderExchangeDTO = useRemoteQueryStep({
+      entry_point: "order_exchange",
+      fields: ["id", "status", "order_id", "return_id"],
+      variables: { id: input.exchangeId },
+      list: false,
+      throw_if_key_not_found: true,
+    })
+
+    const order: OrderDTO = useRemoteQueryStep({
+      entry_point: "orders",
+      fields: ["id", "status", "currency_code"],
+      variables: { id: orderExchange.order_id },
+      list: false,
+      throw_if_key_not_found: true,
+    }).config({ name: "order-query" })
+
+    const shippingOptions = useRemoteQueryStep({
+      entry_point: "shipping_option",
+      fields: [
+        "id",
+        "name",
+        "calculated_price.calculated_amount",
+        "calculated_price.is_calculated_price_tax_inclusive",
+      ],
+      variables: {
+        id: input.shippingOptionId,
+        calculated_price: {
+          context: { currency_code: order.currency_code },
+        },
+      },
+    }).config({ name: "fetch-shipping-option" })
+
+    const shippingMethodInput = transform(
+      { orderExchange, shippingOptions },
+      (data) => {
+        const option = data.shippingOptions[0]
+
+        return {
+          shipping_option_id: option.id,
+          amount: option.calculated_price.calculated_amount,
+          is_tax_inclusive:
+            !!option.calculated_price.is_calculated_price_tax_inclusive,
+          data: option.data ?? {},
+          name: option.name,
+          order_id: data.orderExchange.order_id,
+          return_id: data.orderExchange.return_id,
+          exchange_id: data.orderExchange.id,
+        }
+      }
+    )
+
+    const createdMethods = createOrderShippingMethods({
+      shipping_methods: [shippingMethodInput],
+    })
+
+    const orderChange: OrderChangeDTO = useRemoteQueryStep({
+      entry_point: "order_change",
+      fields: ["id", "status"],
+      variables: { order_id: orderExchange.order_id },
+      list: false,
+    }).config({ name: "order-change-query" })
+
+    validationStep({ order, orderChange })
+
+    const orderChangeActionInput = transform(
+      {
+        orderId: order.id,
+        returnId: orderExchange.return_id,
+        exchangeId: orderExchange.id,
+        shippingOption: shippingOptions[0],
+        methodId: createdMethods[0].id,
+        customPrice: input.customShippingPrice,
+      },
+      ({
+        shippingOption,
+        exchangeId,
+        returnId,
+        orderId,
+        methodId,
+        customPrice,
+      }) => {
+        const methodPrice =
+          customPrice ?? shippingOption.calculated_price.calculated_amount
+
+        return {
+          action: ChangeActionType.SHIPPING_ADD,
+          reference: "order_shipping_method",
+          reference_id: methodId,
+          amount: methodPrice,
+          details: {
```

Comment:
@carlos-r-l-rodrigues, I noticed the add shipping action of Returns and Exchanges differ. Just want to make sure this is intentional.

The return shipping of exchanges does not have `details` as returns. Instead the properties are top-level in the action:
https://github.com/medusajs/medusa/blob/f6e86f1b99abf23cb3b39f15763bd5b2c63c5c53/packages/modules/order/src/services/actions/create-exchange.ts#L237-L244

Reply:
yes, this is correct.
I believe the ideal thing would be having only specific input for each action, and transform to this complex internal object under the hood.
But this would increase a bit the scope

Reply:
So do I put `order_id`, `return_id`, and `claim_id` in the `details` or outside of it?

Shouldn't we be consistent across all types of order changes?

Reply:
Right now, my return shipping order change looks as follows:
```
{
  id: "ordcha_1234",
  details: {
    return_id: "ret_1234",
    order_id: "ord_1234"
  }
  return_id: null,
  order_id: null
}
```

And my exchange return shipping order change as follows:
{
  id: "ordcha_1234",
  details: {}
  return_id: "ret_1234",
  order_id: "ord_1234"
  exchage_id: "exch_1234"
}

Reply:
you can place it only outside details.
I will revisit the other actions and normalise them all.